### ISSUE TITLE
print function

### ISSUE THREAD

**Issue by numapanuma** [gopacifia/DEEP#23]
Be able to print a resource information sheet with its value chain details, current owner/custodian ID, price, other relevant information, product code, etc.

**Comment by escanda** [comment]
I am starting on this one. By print do you mean display or print by a printer a view of this information? In such case, perhaps we might add a button "Display Details" which redirects you to a view with the detailed resource info in which the user can then print the web page (since we are not generating PDFs we can't think for now of "printing" views). Also I would appreciate more details about chain details.

**Comment by fosterlynn** [comment]
@numapanuma will have to speak to the requirement, but in terms of chain details, there is a page called Incoming Value Flow linked from each Resource page that shows that.  There isn't much data in GP's database, but here is an example in Sensorica: http://nrp.sensorica.co/accounting/incoming-value-flows/277/.  It is basically following backwards the Input-Process-Output chain for everything that went into creation of the resource.

**Comment by bhaugen** [comment]
Code for incoming value flows:
https://github.com/gopacifia/DEEP/blob/master/valuenetwork/valueaccounting/models.py#L5340

**Comment by escanda** [comment]
Thanks @fosterlynn and @bhaugen ! Would you please @numapanuma draw me a wireframe of the details page on google draw? It could be huge to make myself an idea of how you want to display the data, such as rectangles with field groups displaying which things and so on. This would help a lot :)

**Comment by numapanuma** [comment]
You're right @escanda, before printing it should be displayable. I didn't think of building a whole wireframe at this point, just the possibility of displaying or exporting to pdf the actual information as it is currently displayed on the page of a given resource.
It will take me some days to propose a wireframe for that. I will share it with you as soon as I get a first draft.

**Comment by escanda** [comment]
Wireframe by @numapanuma: https://docs.google.com/drawings/d/1slKjtBiJfm9h5ZGRu9rtnw2DoQXhakczrH60ysBr2Vw/edit?usp=sharing

**Comment by fosterlynn** [comment]
Wow @numapanuma .  We'll need to piece out what is there and what isn't for @escanda  though.  And figure out how some of that could be implemented from what is there.  Numa, if you have thoughts as to what is what, how these are created and how they are retrieved, that would be helpful. I don't think I have time this week to help figure it out though.  In other words, this just became more than a nice printable page for what we already have on resource.

One thing would be to just lay it out, include the obvious, and leave holding places for the rest.

Or start on another simpler issue for now.

**Comment by bhaugen** [comment]
But it's all relevant info. What DEEP doesn't have, DEEP can get, if the data is available.

**Comment by escanda** [comment]
@fosterlynn those were more or less my initial thoughts, I am pretty new to the code base and writing a reputation system, access rules and so on from the start may not be optimal for the code base itself.

@bhaugen I think as well though these are all interesting approaches to DEEP. But I couldn't know very well how to model them in the REA modelling framework.

BTW Pavel's book is finally on the way to my home so hopefully some of these issues can be solved in a short time.

**Comment by escanda** [comment]
I'll probably go with low hanging fruit issues such as #28 and some more I have to research :)

**Comment by bhaugen** [comment]
@escanda 
>  I couldn't know very well how to model them in the REA modelling framework.

Think of the REA model as a core, and extend other concepts from the related REA concept. REA is meant to be a minimal domain ontology. We always need to extend it with other information from other domains, like location, units of measure, etc.

The DEEP EconomicAgent class has a field for reputation, but we haven't done anything with it. Sensorica wanted a reputation system, and had an affiliate, Bayle Shanks, who was working on one (that never got finished, as far as I know). Lynn and I don't know how to design a good reputation system and are skeptical about them in the first place. I am afraid that people will focus more on polishing their reputations than getting any real work done. I do think that keeping your commitments would be a good input to a reputation system, though.

Also, don't expect Pavel's book to solve all issues, although it is a good book and gets into a lot of REA software details based on a prototype that Pavel and Christian were working on at the time. As I think I mentioned, they have also been rethinking some of it in the past year or two.

**Comment by escanda** [comment]
> Think of the REA model as a core, and extend other concepts from the related REA concept
As a matter of fact I researched about a REA DSL and found about this [DSL](https://publik.tuwien.ac.at/files/PubDat_198185.pdf) by Sonnemberg which was somewhat critized because it diverged from some existing concepts such as commitments to create new similar ones.

> Lynn and I don't know how to design a good reputation system and are skeptical about them in the first place. I am afraid that people will focus more on polishing their reputations than getting any real work done.
Me too I think it might be wiser to qualify transfers based on a criteria such as delivery date, user input and so on; and apply on top a formula. On the other hand I'm a bit lost on reputation systems as well, @numapanuma will probably be able to sketch something more extended in issue #26.

> Also, don't expect Pavel's book to solve all issues
I know. I just need a quickstart into REA so I can afterwards digest more elaborated papers on it :)
